openapi-model/src/main/scala/sttp/apispec/openapi/validation/OpenAPIComparator.scala

```diff
@@ -0,0 +1,249 @@
+package sttp.apispec.openapi.validation
+
+import sttp.apispec.{Schema, SchemaLike}
+import sttp.apispec.openapi.{Header, MediaType, OpenAPI, Operation, Parameter, PathItem, RequestBody, Response}
+import sttp.apispec.validation.SchemaComparator
+
+import scala.collection.immutable.ListMap
+
+class OpenAPIComparator(
+    writerOpenAPI: OpenAPI,
+    readerOpenAPI: OpenAPI
```

Comment:
The "writer" and "reader" distinction makes sense when comparing schemas (which may show up in inputs and outputs), but when comparing OpenAPIs, we should task explicitly about `clientOpenAPI` and `serverOpenAPI`